src/telemetry/base_image.rs

```diff
@@ -0,0 +1,132 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::DeviceManagerError;
+use astarte_sdk::types::AstarteType;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+
+pub fn get_base_image() -> Result<HashMap<String, AstarteType>, DeviceManagerError> {
+    let mut ret: HashMap<String, AstarteType> = HashMap::new();
+
+    let file = BufReader::new(File::open("/etc/os-release")?);
+    let lines = file.lines().flat_map(|line| line);
+    ret.extend(get_from_iter(lines));
+
+    Ok(ret)
+}
+
+fn get_from_iter<I: IntoIterator<Item = String>>(lines: I) -> HashMap<String, AstarteType> {
+    let mut ret: HashMap<String, AstarteType> = HashMap::new();
+    for line in lines {
+        if let Some((key, value)) = line.trim().split_once('=') {
+            match key {
+                "IMAGE_ID" => {
+                    let value = value.replace('"', "");
+                    ret.insert("/name".to_string(), AstarteType::String(value));
+                }
+                "IMAGE_VERSION" => {
+                    let value = value.replace('"', "");
+                    if let Some((version, build_id)) = value.split_once('+') {
```

Comment:
we can avoid this by using: https://docs.rs/semver/latest/semver/
Anyway so far it looks ok.